plato/draw/Box.py

```diff
@@ -0,0 +1,100 @@
+import itertools
+import numpy as np
+from .Lines import Lines
+from .. import math
+from .internal import ShapeDecorator, ShapeAttribute
+
+@ShapeDecorator
+class Box(Lines):
+    """A triclinic box frame.
+
+    This primitive draws a triclinic box centered at the origin. It infers
+    dimensionality and can be used to draw a 3D or 2D box, based on whether
+    :code:`Lz` is zero. The box argument must be a tuple or list of length
+    2, 3, or 6, an object with attributes :code:`Lx, Ly, Lz, xy, xz, yz`, or
+    a dictionary with those keys. The values of :code:`Lz, xy, xz, yz` are zero
+    by default.
+
+    Examples::
+        Lx = Ly = Lz = 10
+        xy = xz = yz = 0
+        box_primitive = draw.Box((Lx, Ly, Lz, xy, xz, yz), width, color)
+    """
+
+    _ATTRIBUTES = Lines._ATTRIBUTES + list(
+        itertools.starmap(ShapeAttribute, [
+            ('box', np.float32, (1, 1, 1, 0, 0, 0), 1, False,
+             'Box object'),
+            ('width', np.float32, 0.01, 0, False,
+             'Width of box line segments'),
+            ('color', np.float32, (0, 0, 0, 1), 1, False,
+             'Color, RGBA, [0, 1] for the box line segments'),
+        ]))
+
+    @staticmethod
+    def _duck_type_box(box):
+        """Duck type the box from a valid input.
+
+        Boxes can be a list, dictionary, or object with attributes.
+        """
+        try:
+            # Handles freud.box.Box and namedtuple
+            Lx = box.Lx
+            Ly = box.Ly
+            Lz = getattr(box, 'Lz', 0)
+            xy = getattr(box, 'xy', 0)
+            xz = getattr(box, 'xz', 0)
+            yz = getattr(box, 'yz', 0)
+        except AttributeError:
+            try:
+                # Handle dictionary-like
+                Lx = box['Lx']
+                Ly = box['Ly']
+                Lz = box.get('Lz', 0)
+                xy = box.get('xy', 0)
+                xz = box.get('xz', 0)
+                yz = box.get('yz', 0)
+            except (IndexError, KeyError, TypeError):
+                if not len(box) in [2, 3, 6]:
+                    raise ValueError(
+                        "List-like objects must have length 2, 3, or 6 to be "
+                        "converted to a box.")
+                # Handle list-like
+                Lx = box[0]
+                Ly = box[1]
+                Lz = box[2] if len(box) > 2 else 0
+                xy, xz, yz = box[3:6] if len(box) == 6 else (0, 0, 0)
+        return (Lx, Ly, Lz, xy, xz, yz)
+
+    def __init__(self, *args, **kwargs):
```

Comment:
The initializer here just rewrites `kwargs` so that they will properly feed into the constructor for the parent class `Lines` (or the appropriate backend's child class).